lib/minheap.js

```diff
@@ -38,26 +46,50 @@ class MinHeap {
   }
 
   // This method returns true if the heap is empty
-  // Time complexity: ?
-  // Space complexity: ?
+  // Time complexity: O(1)
+  // Space complexity: O(1)
   isEmpty() {
-    throw new Error("Method not implemented yet...");
+    return this.store.length === 0;
   }
 
   // This helper method takes an index and
   //  moves it up the heap, if it is less than it's parent node.
   //  It could be **very** helpful for the add method.
-  // Time complexity: ?
-  // Space complexity: ?
+  // Time complexity: O(log n)
+  // Space complexity: O(1)
   heapUp(index) {
-    throw new Error("Method not implemented yet...");
+    let index1 = index;
+    let parentIndex = Math.floor((index - 1)/2);
+
+    while (parentIndex >= 0 && this.store[parentIndex].key > this.store[index1].key) {
+      this.swap(index1, parentIndex);
+      index1 = parentIndex;
+      parentIndex = Math.floor((index1 - 1)/2);
+    }
```

Comment:
Here's another approach that would let us perform the `parentIndex` calculation only once (we can get rid of the once before the start of the loop). It also allows us to simplify the `while` condition a bit.

```js
    while (index > 0) {
      const parentIndex = Math.floor((index - 1)/2);
      
      if (this.store[parentIndex].key < this.store[index].key) { break; }

      this.swap(index, parentIndex);
      index = parentIndex;
    }
```